src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs

```diff
@@ -92,10 +92,9 @@ protected ImmutableArray<ISymbol> GetSymbols(IParameterSymbol parameter, int pos
         private ImmutableArray<ITypeSymbol> GetTypeSymbols(ImmutableArray<ISymbol> candidateSymbols, int ordinalInInvocation, int ordinalInLambda)
         {
             var expressionSymbol = _context.SemanticModel.Compilation.GetTypeByMetadataName(typeof(Expression<>).FullName);
-            var funcSymbol = _context.SemanticModel.Compilation.GetTypeByMetadataName(typeof(Func<>).FullName);
             // We cannot help if semantic model is corrupted or incomplete (e.g. during solution loading) and those symbols are not loaded. 
             // Just bail out.
-            if (expressionSymbol == null || funcSymbol == null)
+            if (expressionSymbol == null)
```

Comment:
I would assume we would skip the special case around Expression<> but the regular delegate handling should work just fine. I'd be more worried if there's some framework out there people target that doesn't have expression tree support.

Reply:
Do you see more issues here?

---
In reply to: [276405774](https://github.com/dotnet/roslyn/pull/35067#discussion_r276405774) [](ancestors = 276405774)

Reply:
My worry is we're throwing out the entire behavior change if we can't find Expression, which I could imagine happens in some older portable frameworks or something. I'd just say remove the check here, and add a `expressionSymbol != null` check in your if block after the `If type is <see cref="Expression{TDelegate}"/>, ignore <see cref="Expression"/> and use TDelegate.` comment.

Reply:
Cool! Great idea! Thank you!